lgc/patch/PatchBufferOp.cpp

```diff
@@ -426,6 +434,9 @@ void PatchBufferOp::visitCallInst(CallInst &callInst) {
     m_replacementMap[&callInst] = std::make_pair(nullptr, nullptr);
 
     callInst.replaceAllUsesWith(bufferLength);
+
+    m_descriptorUsers.insert(bufferLength);
```

Comment:
We don't need to save a list of places to post-process. The code for handling a null descriptor in a length operation can just be incorporated here. No need for m_descriptorUsers at all. You might still like to put the code to check whether the descriptor is null in a separate function, as it looks like you are going to use that again in future work.

```
Value *desc = m_replacementMap[pointer].first;
Value *bufferLength = m_builder->CreateExtractElement(desc, 2);
if (m_pipelineState->getPipelineOptions().robustAccess) {
  Value *isNullDesc = createIsNullDesc(desc); // using m_builder
  bufferLength = m_builder->CreateSelect(isNullDesc, m_builder->getInt32(0), bufferLength);
}
callInst.replaceAllUsesWith(bufferLength);
```

Reply:
Yes. I need to handle store and load in future work. To skip store operation by creating branching block, I need to do it in a post-process. So I decided to put all the code in a seperate function. Do you suggest not gathering them, such as handling the lenghth and load operation in visit function and store in post-process?

Reply:
Yes, I think it is better not to gather them. Even the store can be handled in the visit function I think. For creating the if around the store, BuilderImpl::createIf might work, but you would have to move it to BuilderBase so you can access it from outside builder.

Reply:
Thanks. I will do it as you suggested.